std/selector/multiplexer.go

```diff
@@ -53,12 +55,41 @@ func Map(api frontend.API, queryKey frontend.Variable,
 // sel needs to be between 0 and n - 1 (inclusive), where n is the number of
 // inputs, otherwise the proof will fail.
 func Mux(api frontend.API, sel frontend.Variable, inputs ...frontend.Variable) frontend.Variable {
-	// we use BinaryMux when len(inputs) is a power of 2.
-	if binary.OnesCount(uint(len(inputs))) == 1 {
-		selBits := bits.ToBinary(api, sel, bits.WithNbDigits(binary.Len(uint(len(inputs)))-1))
+	n := uint(len(inputs))
+	nbBits := binary.Len(n - 1)
+	selBits := bits.ToBinary(api, sel, bits.WithNbDigits(nbBits)) // binary decomposition ensures sel < 2^nbBits
```

Comment:
When `len(inputs) == 1` then this fails. Should handle separately.

Reply:
I think its better to have the assertion for consistency with case len(inputs)>1

Reply:
Ok, edited as suggested.